cmd/cainjector/app/start.go

```diff
@@ -35,10 +36,9 @@ import (
 	_ "k8s.io/client-go/plugin/pkg/client/auth"
 	"k8s.io/client-go/tools/leaderelection/resourcelock"
 	"k8s.io/component-base/logs"
-	logsapi "k8s.io/component-base/logs/api/v1"
-	_ "k8s.io/component-base/logs/json/register"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
```

Comment:
Why do we rely on component-base's `logsapi.AddFlags` in some places (controller, webhook) but we rely on controller-runtime's `config.RegisterFlags` in some other places (cainjector)?

Reply:
Both `logf.AddFlags` and `config.RegisterFlags` are required:
- `logf.AddFlags` adds the logging flags
- `config.RegisterFlags` adds the kubeconfig flag (which is configured by controller-runtime)

The reason that we need `config.RegisterFlags` now is because I removed the `cmd.Flags().AddGoFlagSet(flag.CommandLine)` line on https://github.com/cert-manager/cert-manager/pull/5828/files#diff-dae145745f14d27c7924b0bdd2de92eee62f58264fbcd938ca2689106c8db923L45.

Reply:
Thanks. Why did you change

```go
cmd.Flags().AddGoFlagSet(flag.CommandLine)
```

to

```go
config.RegisterFlags(flag.CommandLine)
```
?

Reply:
The code looks like this now:
https://github.com/cert-manager/cert-manager/pull/5828/files/d73c8bf150ecc1328184642f35f9c05bea89bb7f..31fa98f6bdbae5c52801422e1187dc7329f7e32b#diff-02edd2afed286bcf684fd4df60744cffd39dc302acc3ab8b863a30df799cfcfeR137-R145

I'm not calling config.RegisterFlags on flag.CommandLine, instead I created a temporary flagset that adds the flags to the cobra cmd. My main goal is to be a lot more explicit about registering flags. If we just use the global flag.CommandLine value, any imported library can start registering flags and that makes it more complicated to know where flags are comming from.